volatility3/framework/layers/intel.py

```diff
@@ -264,12 +269,41 @@ def _mapping(self,
                 length -= chunk_size
                 offset += chunk_size
 
+            if offset > self.maximum_address:
+                break
+            
+            offset = self.get_canonical_address(offset)
+
     @property
     def dependencies(self) -> List[str]:
         """Returns a list of the lower layer names that this layer is dependent
         upon."""
         return [self._base_layer] + self._swap_layers
 
+    @classmethod
+    def is_address_canonical(cls, address: int) -> bool:
+        """Returns whether the address is canonical or not.
+        When a 64-bit linear address is generated to access memory, the processor first
+        confirms that the address is canonical. If the address is not canonical, the memory
```

Comment:
Reference please.

Reply:
Same as above.
The LAM57 and LAM48 comments are taken from the `Intel® Architecture Instruction Set Extensions Programming Reference` - `CHAPTER 14
LINEAR ADDRESS MASKING (LAM)`.